metadata/metadata.go

```diff
@@ -54,6 +54,67 @@ func AddLabels(obj v1.Object, entries map[string]string) error {
 	return nil
 }
 
+// CopyAnnotationsByPrefix copies all annotations from a source object to a destination object where the key matches
+// the specified sourcePrefix.
+func CopyAnnotationsByPrefix(source, destination v1.Object, prefix string) error {
+	if source == nil || destination == nil {
+		return errors.New("object cannot be nil")
+	}
+
+	if destination.GetAnnotations() == nil {
```

Comment:
Let's remove this and move it to the auxiliary function as commented.

Reply:
It can't pass the test when I tried to move it there - It("should copy all the labels matching the given prefix when Labels in destination is empty", func() {
                        podSource := &corev1.Pod{
                                ObjectMeta: v1.ObjectMeta{
                                        Labels: map[string]string{"foo": "bar", "baz": "qux"},
                                },
                        }
                        podDest := &corev1.Pod{
                                ObjectMeta: v1.ObjectMeta{},
                        }
                        CopyLabelsByPrefix(podSource, podDest, "fo")
                        Expect(podDest.Labels).To(gstruct.MatchAllKeys(gstruct.Keys{
                                "foo": Equal("bar"),
                        }))
                })

Reply:
If we pass nil to destination for the func copyWithPrefixReplacement(source, destination map[string]string, sourcePrefix, destinationPrefix string) , it make sense to  keep the destination as nil after the function is run. So, I think the current code is ok for me.